Framework/DataObjects/inc/MantidDataObjects/WorkspaceCreation.h

```diff
@@ -175,19 +192,40 @@ std::unique_ptr<T> create(const IndexArg &indexArg, const HistArg &histArg) {
   return std::move(ws);
 }
 
+template <class T, class IndexArg,
+          typename std::enable_if<
+              !std::is_base_of<API::MatrixWorkspace, IndexArg>::value>::type * =
+              nullptr>
+std::unique_ptr<T> create(const IndexArg &indexArg,
+                          const HistogramData::Histogram &histogram) {
+  auto ws = Kernel::make_unique<T>();
+  HistogramData::Histogram histogramTemplate(histogram);
+  if (std::is_base_of<DataObjects::EventWorkspace, T>::value) {
+    histogramTemplate = detail::stripData(histogramTemplate);
+  }
+  ws->initialize(indexArg, histogramTemplate);
+  return std::move(ws);
+}
+
 template <class T, class P,
           typename std::enable_if<std::is_base_of<API::MatrixWorkspace,
                                                   P>::value>::type * = nullptr>
 std::unique_ptr<T> create(const P &parent) {
-  return create<T>(parent, parent.getNumberHistograms(),
-                   detail::stripData(parent.histogram(0)));
-}
+  const auto numHistograms = parent.getNumberHistograms();
+  if (parent.isCommonBins()) {
```

Comment:
I think the `isCommonBins` method is a bit broken, I would not use it. Simply remove this `if` and set up sharing X below, that should be both safer and faster.